### PR TITLE
fixed short circuit expressions, optimized, cleanup

### PR DESCRIPTION
**Pull request by colinsurprenant**
disclaimer: sorry for whitespace diffs :(

- fixed the broken short circuit evaluation per #19
- optimized code but removing useless checks and avoiding superfluius regex object creation
- overall code cleanup
- added specs to validate short circuit fonctionality

### CODE REVIEW

File: spec/filters/kv_spec.rb [hunk 1]
```diff
@@ -54,21 +54,108 @@
...
 
-    sample "hello:=world foo:bar baz=:fizz doublequoted:\"hello world\" singlequoted:'hello world' brackets:(hello world)" do
-      insist { subject["hello"] } == "=world"
-      insist { subject["foo"] } == "bar"
-      insist { subject["baz="] } == "fizz"
-      insist { subject["doublequoted"] } == "hello world"
-      insist { subject["singlequoted"] } == "hello world"
-      insist { subject["brackets"] } == "hello world"
+      sample "hello:=world foo:bar baz=:fizz doublequoted:\"hello world\" singlequoted:'hello world' brackets:(hello world)" do
+        insist { subject["hello"] } == "=world"
```
Comment (untergeek) [inline spec/filters/kv_spec.rb]: Replace the `insist`s with `expect`s?

I mean, while the opportunity is here...

**Comment by untergeek** [comment]
With the exceptions of the `insist`s, LGTM™

The extra tests are a welcome addition, and the code is much easier to read now.

Tests pass for me.

File: spec/filters/kv_spec.rb (hunk 1 above)
Comment (colinsurprenant) [inline spec/filters/kv_spec.rb]: yeah, well, the specs is quite large as it is and I didn't want to do it all now :P

**Comment by colinsurprenant** [comment]
thanks @untergeek, will merge!

**Comment by jordansissel** [comment]
Code LGTM

**Comment by colinsurprenant** [comment]
squashing, merging and will bump version in PR #21

**Comment by colinsurprenant** [comment]
changes pushed in version 2.0.3 per PR #21

**Pull request merged**